File: Difficulty: Easy/Anagram/anagram.java
class Solution {
    public static boolean areAnagrams(String s1, String s2) {
        // code here
        if(s1.length()!=s2.length()){
            return false;
        }
        
        int[] freq = new int[26];
        for(char ch:s1.toCharArray()){
            freq[ch-'a']++;
        }
        for(char ch:s2.toCharArray()){
            freq[ch-'a']--;
        }
        for(int c : freq){
            if(c!=0){
                return false;
            }
        }
        return true;
    }
}

File: Difficulty: Easy/Anagram/anagram.cpp
class Solution {
  public:
    bool areAnagrams(string& s1, string& s2) {
        // code here
        if(s1.size()!=s2.size()){
            return false;
        }
        unordered_map<char,int> freq;
        for(char ch:s1){
            freq[ch]++;
        }
        for(char ch:s2){
            freq[ch]--;
        }
        for(auto& [c,count]:freq){
            if(count!=0){
                return false;
            }
            
        }
        return true;
    }
};